plugins/fs/src/commands.rs

```diff
@@ -23,13 +23,43 @@ use std::{
 
 use crate::{scope::Entry, Error, FilePath, FsExt};
 
-#[derive(Debug, serde::Deserialize)]
-#[serde(untagged)]
+#[derive(Debug)]
 pub enum SafeFilePath {
     Url(url::Url),
     Path(SafePathBuf),
 }
 
+impl<'de> serde::Deserialize<'de> for SafeFilePath {
```

Comment:
I see that we have a duplicated enum definition with and now same deserialize implementation, maybe we can combine this logic into one, by using generics? and still export aliases without a generic:
```rs 
enum FilePathOrUrl<T: FromStr> {
    Url(url::Url),
    Path(T),
}
pub type SafeFilePath = FilePath<SafeFilePath>;
pub type FilePath = FilePath<PathBuf>;
`

Reply:
> ```rust
> enum FilePathOrUrl<T: FromStr> {
>     Url(url::Url),
>     Path(T),
> }
> pub type SafeFilePath = FilePath<SafeFilePath>;
> pub type FilePath = FilePath<PathBuf>;
> ```

I'm not sure how that'll work, `SafeFilePath` doesn't implement `FromStr`, and that means we either need to do it in tauri or we need a wrapper

Reply:
The best is to implement `FromStr` on in tauri. For now, we can just add a TODO then or merge as is.